test/example/CMakeLists.txt

```diff
@@ -1,8 +1,30 @@
-add_executable(example.test)
-target_sources(example.test PRIVATE example.test.cxx)
-target_link_libraries(example.test PRIVATE example::example)
+# Setup the `lit` tool
+add_custom_command(
+    OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/venv/bin/lit"
+    COMMAND python3 -m venv "${CMAKE_CURRENT_BINARY_DIR}/venv"
+    COMMAND "${CMAKE_CURRENT_BINARY_DIR}/venv/bin/pip" install --upgrade pip
+    COMMAND "${CMAKE_CURRENT_BINARY_DIR}/venv/bin/pip" install --upgrade lit
+)
+
+# Setup the test suite configuration
+configure_file("${CMAKE_SOURCE_DIR}/test/support/lit.cfg.in"
+               "${CMAKE_CURRENT_BINARY_DIR}/lit.cfg")
+
+add_custom_target(test-depends
+    COMMAND true
+    DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/venv/bin/lit"
+            "${CMAKE_CURRENT_BINARY_DIR}/lit.cfg"
+            example
+    COMMENT "Setup the test dependencies"
+)
+
+add_test(
+    NAME setup-tests
+    COMMAND "${CMAKE_COMMAND}" --build "${CMAKE_BINARY_DIR}" --target test-depends
+)
```

Comment:
This is awkward. It looks like a CTest test cannot depend on a `add_custom_target`, so I need to do things in this super roundabout way to ensure that the test suite is set up before the tests actually start running.

Reply:
There is no guarantee that ctest will run `setup-tests` before `example.test`. It might do that by default but there is a ctest option to run the tests in a random order. Consider dropping the `setup-tests` test and renaming the `test-depends` target `example.test`. This would then emulate the usual pattern followed when the test is an actual executable and not a call to a python script.

```
$ cmake --build --target example.test
$ ctest -R example.test
```

But then again, that might be misleading a user to think the `example.test` target is an executable. I agree, it's awkward.

Reply:
CMake doesn't provide a way to indicate tests have dependencies so the prevailing practice is for tests to assume all targets part of the default target are built.

To ensure `test-depends` is part of the default target, you need only add the `ALL` keyword to the `add_custom_target` call.